ruby/treating_arrays_as_sets.md

```diff
@@ -0,0 +1,39 @@
+# Treating arrays as sets
+
+Even though Ruby has a [set implementation in its standard
+library](http://ruby-doc.org/stdlib-2.2.2/libdoc/set/rdoc/Set.html), you might
+often find yourself wanting to use a regular array as a set. For example, you
+might want to push an element into an array without introducing duplicates:
+
+```ruby
+things = [x, y]
+things << z unless things.include?(z)
+```
+
+However, arrays implement several set operations themselves:
+
+* [set intersection](https://en.wikipedia.org/wiki/Intersection_(set_theory)) via [`&`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-26);
+* [set difference](https://en.wikipedia.org/wiki/Complement_(set_theory)) via [`-`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-2D);
+* [set union](https://en.wikipedia.org/wiki/Union_(set_theory)) via [`|`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-7C).
+
+You can take advantage of these to avoid pushing duplicates like so:
+
+```ruby
+things = [x, y]
+things |= [z]
+```
+
+This works because it is functionally equivalent to:
+
+```ruby
+things = [x, y]
+things = things | [z]
+```
+
+You can also use intersection in the same way to retrieve only common
+elements:
+
+```ruby
+things = [x, y]
+things &= [z]
```

Comment:
I'd be more inclined to highlight the use of `-=` for removal of items from a set, since that's more likely to be of use.

And I agree with @jakubpawlowicz – the use of `&=` to mutate an array's values is probably best warned about, if anything. While there may be use cases for it, I'd say they're unlikely in the context of set-like usage.